utils/docker.go

```diff
@@ -45,15 +46,13 @@ func DockerLogin(username, password, email string) (string, error) {
 			Password:      password,
 			ServerAddress: registry.IndexServerAddress(),
 		}
-		status, err := registry.Login(&authconfig, registry.HTTPRequestFactory(nil))
+		status, err = registry.Login(&authconfig, registry.HTTPRequestFactory(nil))
 		if err != nil {
 			return "", err
 		}
-		return status, nil
 	}
 
-	return "", fmt.Errorf("Auth params don't make sense.")
-
+	return status, nil
```

Comment:
So, this code needs some fixing - that last return of mine is pretty laughable. But what about this? (winging it, this might be broken)

```
func DockerLogin(username, password, email string) (string, error) {
	var status string
	var err error

	if username == "" && password == "" && email == "" {
		// Attempt login with .dockercfg file.
		configFile, err := registry.LoadConfig(os.Getenv("HOME"))
		if err != nil {
			return "", err
		}
		authconfig, ok := configFile.Configs[registry.IndexServerAddress()]
		if !ok {
			return "", fmt.Errorf("Error: Unable to login, no data for index server.")
		}
		status, err = registry.Login(&authconfig, registry.HTTPRequestFactory(nil))
		if err != nil {
			return "", err
		}
		return status, nil
	} 
	// Attempt login with this function's auth params.
	authconfig := registry.AuthConfig{
		Username:      username,
		Email:         email,
		Password:      password,
		ServerAddress: registry.IndexServerAddress(),
	}
	status, err = registry.Login(&authconfig, registry.HTTPRequestFactory(nil))
	if err != nil {
		return "", err
	}
	return status, nil
}
```

Then if we're looking at a traceback, it's more obvious which path was taken. I'm very open to discussion.